node/utils/metrics/auth.ts

```diff
@@ -0,0 +1,43 @@
+import type { Logger } from '@vtex/api/lib/service/logger/logger'
+
+import type { Metric } from './metrics'
+import { B2B_METRIC_NAME, sendMetric } from './metrics'
+
+export interface AuthAuditMetric {
+  operation: string
+  forwardedHost: string
+  caller: string
+  role: string
+  permissions: string[]
+  hasAdminToken: boolean
+  hasStoreToken: boolean
+  hasApiToken: boolean
```

Comment:
Do we care which of these tokens its being used? Maybe we should emit just a `hasToken` field aggregating all of these to reduce the amount of data we emit?

Reply:
The intention was to monitor which token was used for each operation. I didn't use just one because I'm not sure if we should allow multiple tokens in some cases.